pkg/internal/request/span.go

```diff
@@ -0,0 +1,72 @@
+package request
+
+import (
+	"time"
+
+	"github.com/gavv/monotime"
+)
+
+type EventType int
+
+const (
+	EventTypeHTTP EventType = iota + 1
+	EventTypeGRPC
+	EventTypeHTTPClient
+	EventTypeGRPCClient
+)
+
+type converter struct {
+	clock     func() time.Time
+	monoClock func() time.Duration
+}
+
+var clocks = converter{monoClock: monotime.Now, clock: time.Now}
+
+// Span contains the information being submitted by the following nodes in the graph.
+// It enables confortable handling of data from Go.
+type Span struct {
+	Type          EventType
+	ID            uint64
+	Method        string
+	Path          string
+	Route         string
+	Peer          string
+	Host          string
+	HostPort      int
+	Status        int
+	ContentLength int64
+	RequestStart  int64
+	Start         int64
+	End           int64
+	ServiceName   string
+	Metadata      []MetadataTag
+}
```

Comment:
I think this is going to need "TraceID string" after the changes I just merged.

Reply:
Done!